azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java

```diff
@@ -1074,6 +1077,11 @@ private boolean isServiceRequired() {
    * @throws ExecutorManagerException
    */
   private void deletePod(final int executionId) throws ExecutorManagerException {
+    final ExecutableFlow flow = this.executorLoader.fetchExecutableFlow(executionId);
```

Comment:
Callers of `deletePod` likely already have access to the `ExecutableFlow` of the object, could you please check if this additional call to `fetchExecutableFlow` can be avoided?

Reply:
I believe deleteFlowContainer has a reference to the executionId but not the ExecutableFlow as it is reliant on the AzStatusMetadata.